dj/service_clients.py

```diff
@@ -130,3 +131,27 @@ def get_query(
             )
         query_info = response.json()
         return QueryWithResults(**query_info)
+
+    def materialize_cube(  # pylint: disable=too-many-arguments
+        self,
+        node_name: str,
+        schedule: str,
+        query: str,
+        spark_conf: str,
+        druid_spec: Dict,
+    ):
+        """
+        Kick off scheduling of materialization job for cube nodes
+        """
+        response = self.requests_session.post(  # pragma: no cover
+            "/materialization/druid/",
```

Comment:
Should this be hardcoded to druid? Since the json in the request is pretty permissive (it can contain a spark conf, druid spec, and I'm guessing more as more implementations of `MaterializationJob` are added), I would have thought it would all go to a generic `/materialization/` endpoint on the query service.

Reply:
Yeah, so in this case the function is `materialize_cube` and it's meant to be only for cubes, but it's true that it's not limited to Druid. I went ahead and removed `druid` and also added a `node_type` parameter so that we can tell the query service what the type of materialization is. For example, in the Druid cube materialization, it has both Spark and Druid config because we use Spark to create the table first, then we ingest the table into Druid.